### DISCUSSION TITLE
How to specify a vector database

### DISCUSSION THREAD

**Discussion by rudadasir** [langgenius/dify#4308] [Help] [answered]
when self-hosting,do we need to specify a vector databae or it is automatically?

**Comment by patryk20120** [comment, marked as answer]
In the docker-compose.yaml file (https://github.com/langgenius/dify/blob/main/docker/docker-compose.yaml) you can specify what Vector DB you want to use (default: `weaviate`, available: `weaviate`, `qdrant`, `milvus`, `relyt`) - please remember to uncomment container if you choose not the default DB. You have to also change `VECTOR_STORE` env https://github.com/langgenius/dify/blob/main/docker/docker-compose.yaml#L93

**Reply by LagisettyRavikiran** [reply]
Hello, In docker I run the docker-compose.yaml it is not pulling the milvus DB instance. How to pull that image in the dify

**Comment by NegTech** [comment]
isnt it possible to change the vector DB in cloud version ?
